indra/newview/llpanelface.cpp

```diff
@@ -1132,65 +1107,51 @@ void LLPanelFace::updateUI(bool force_set_values /*false*/)
         mCtrlColorTransp->setValue(editable ? transparency : 0);
         mCtrlColorTransp->setEnabled(editable && has_material);
 
+        // Shiny
         U8 shiny = 0;
-        bool identical_shiny = false;
+        {
+            bool identical_shiny = false;
 
-        // Shiny
-        LLSelectedTE::getShiny(shiny, identical_shiny);
-        identical = identical && identical_shiny;
+            LLSelectedTE::getShiny(shiny, identical_shiny);
+            identical = identical && identical_shiny;
 
-        shiny = specmap_id.isNull() ? shiny : SHINY_TEXTURE;
+            shiny = specmap_id.isNull() ? shiny : SHINY_TEXTURE;
 
-        LLCtrlSelectionInterface* combobox_shininess = childGetSelectionInterface("combobox shininess");
-        if (combobox_shininess)
-        {
-            combobox_shininess->selectNthItem((S32)shiny);
-        }
+            mComboShininess->getSelectionInterface()->selectNthItem((S32)shiny);
 
-        getChildView("label shininess")->setEnabled(editable);
-        getChildView("combobox shininess")->setEnabled(editable);
+            mLabelShininess->setEnabled(editable);
+            mComboShininess->setEnabled(editable);
 
-        getChildView("label glossiness")->setEnabled(editable);
-        getChildView("glossiness")->setEnabled(editable);
+            mLabelGlossiness->setEnabled(editable);
+            mGlossiness->setEnabled(editable);
 
-        getChildView("label environment")->setEnabled(editable);
-        getChildView("environment")->setEnabled(editable);
-        getChildView("label shinycolor")->setEnabled(editable);
+            mLabelEnvironment->setEnabled(editable);
+            mEnvironment->setEnabled(editable);
+            mLabelShiniColor->setEnabled(editable);
 
-        getChild<LLUICtrl>("combobox shininess")->setTentative(!identical_spec);
-        getChild<LLUICtrl>("glossiness")->setTentative(!identical_spec);
-        getChild<LLUICtrl>("environment")->setTentative(!identical_spec);
-        mShinyColorSwatch->setTentative(!identical_spec);
+            mComboShininess->setTentative(!identical_spec);
+            mGlossiness->setTentative(!identical_spec);
+            mEnvironment->setTentative(!identical_spec);
+            mShinyColorSwatch->setTentative(!identical_spec);
```

Comment:
Please move all there changes to a separate commit.

Not only they are massive and make checking your review troublesome, in case of any issues tracking changes or reverting will be hard.